pynest/nest/lib/hl_api_helper.py

```diff
@@ -424,63 +454,43 @@ def model_deprecation_warning(model):
         warnings.warn('\n' + text)
 
 
-def turn_off_deprecation_warning(deprecated_func_name):
-    """Turns off deprecation warnings on SLI and Python level for given
-    function.
+class SuppressedDeprecationWarning(object):
```

Comment:
I think this should not be exposed to users, so maybe add a ``_`` before and a ``__all__`` in the file?